Source/FullScreenPokemon.ts

```diff
@@ -4829,6 +4829,63 @@ module FullScreenPokemon {
                 callback);
         }
 
+        /* partyActivate functions
+         */
+
+        /**
+         * Makes sure that Player is facing the correct HMSolid
+         *
+         * @param player   The Player.
+         * @param pokemon   The Pokemon using the move.
+         * @param move   The move being used.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         *       Also need to add context for what happens if player is not bordering the correct HMSolid.
+         */
+        partyActivateCheckThing(player: IPlayer, pokemon: IPokemon, move: IMoveSchema): void {
+            if (player.bordering[player.direction]) {
+                if (player.bordering[player.direction] === move.solid) {
+                    move.partyActivate(player, pokemon);
+                }
+            }
+        }
+
+        /**
+         * Makes sure the Player has a Pokemon in the party that knows the necessary HM move.
+         * 
+         * @param player   The Player.
+         * @param thing   The Solid to be affected.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         *       Also need to add functionality to close the menu if the Pokemon has the move.
+         */
+        checkPartyActivate(player: IPlayer, thing: IHMSolid): void {
+            var partyPokemon: IPokemon[] = player.FSP.ItemsHolder.getItem("PokemonInParty"),
+                moves: BattleMovr.IMove[],
+                i: number,
+                j: number;
+
+            for (i = 0; i < partyPokemon.length; i += 1) {
+                moves = partyPokemon[i].moves;
+
+                for (j = 0; j < moves.length; j += 1) {
+                    if (moves[j].title === thing.moveName) {
+                        thing.moveCallback(player, partyPokemon[i]);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Cuts a cuttable tree.
+         *
+         * @param player   The Player.
+         * @param pokemon   The Pokemon using Cut.
+         * @returns True if tree can be cut. 
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         *       Also need to add a Cutscene for what happens when the CuttableTree is cut.
```

Comment:
No need for an entire cutscene; this isn't part of one. Just say "an animation".

Also, since these are two todos, put a separate `@todo` tag for the second line.